adapter-guide/develop-stix-adapter.md

```diff
@@ -0,0 +1,602 @@
+
+*Create Query Connection*
+
+```
+def create_query_connection(self, query): 
+      return { "success": True, "search_id": query }
+```      
+
+*Create Status Connection*
+```
+def create_status_connection(self, search_id): 
+      return {"success": True, "status": "COMPLETED", "progress": 100}
+```
+
+#### Step 4. Verify that the transmission module was created successfully
+{: #transmission-4}
+
+1.	You must have:
+    * Authentication credentials to connect to the data source.
+    * A private key and certificate for the cert.
+
+    Authentication depends on the data source and can be: 
+    ```
+    '{"auth": {"username": "<username>","password": "<password>"}}' or
+    '{"auth": {"SEC":"<SEC TOKEN>"}}'
+    ```
+
+2.	Test the transmission ping method.  
+    1. Use the following CLI command:    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth": <authentication object>}' ping
+        ```
+    
+    2. Visually confirm that a result comes back with 
+        ```
+        {'success': True}
+        ```
+
+3.	Test the transmission is_async method.  
+    1. Use the following CLI command:    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth{"auth": <authentication object>}' is_async
+        ```
+    2. Visually confirm that it returns true if the data source is asynchronous. Otherwise, it must return false.
+
+4.	Test the transmission query method.  
+    1. Use the following CLI command:    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth": <authentication object>}' query "<Native data source Query String>"
+        ```
+
+    2. Visually confirm that a result comes back with 
+        ```
+        {'success': True, 'search_id': '<some query UUID>'}
+        ```
+
+    3. Take note of the UUID that is returned. It is the ID to use in the rest of the tests.
+
+5.	Test the transmission status method.  
+    1. Use the following CLI command:    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth{"auth": <authentication object>}' status "<Query UUID from test C>"
```

Comment:
Should now refer to test 4 rather than test C given the change in number of the tests.